tools/appveyor.psm1

```diff
@@ -517,16 +517,23 @@ function Push-Artifact
     param(
         [Parameter(Mandatory)]
         [ValidateScript({Test-Path -Path $_})]
-        $Path
+        $Path,
+        [string]
+        $Name
     )
 
-    if($env:Appveyor)
+    if(!$Name)
+    {
+        $artifactName = [system.io.path]::GetFileName($Path)
+    }
+    else
     {
-        Push-AppveyorArtifact $Path
+        $artifactName = $Name
     }
-    elseif ($env:TF_BUILD -and $env:BUILD_REASON -ne 'PullRequest') {
+
+    if ($env:TF_BUILD) {
```

Comment:
If `$env:TF_BUILD` is not defined then this cmdlet does nothing.  Is that Ok, or should a warning be generated?